### PR TITLE
add shouldPrefersStatusBarHidden, Fixed #332, #333

### LINKED ISSUE #332: [strong bug] Error in isUserInteractionEnabled (Please treat preferentially)
**Describe the bug**
Although i have reported many bugs for some days, i really like this framework 👍 !! 

Well, I noticed that the UIBarButtonItem sometimes did not work in the UiNavigationBar (because, userInteractionEnabled = false). I did not understand why and it really annoyed me. Now I have found the reason why it is.

**To Reproduce**
1. Look down at my small example. >> RUN
2. First, hide the NavBar by scrolling down
2. If you just scroll up so far that the NavBar is halfway visible, then you lift the finger off. The scrollbar now expands by itself. The status of userInteractionEnabled is now false. So you can not tap the button on UiNavigationBar. 

The bug was created at:

` navigationBar.isUserInteractionEnabled = (newState == .expanded)`

on line 448 at ScrollingNavigationController.swift

**Expected behavior**
The UIBarButtonItems should always have (userInteractionEnabled = true) after finished scrolling

**Example GIT**
https://github.com/StefaniOSApps/Example

### PR DESCRIPTION
**Pull request by StefaniOSApps**
Now, you can set prefersStatusBarHidden from the ScrollingNavigationController.
Defaults to `false`

### CODE REVIEW

**Comment by StefaniOSApps** [comment]
Fixed #332, #333 at showNavbar, hideNavbar

File: Source/ScrollingNavigationController.swift [hunk 1]
```diff
@@ -131,7 +131,18 @@ open class ScrollingNavigationController: UINavigationController, UIGestureRecog
    
    Defaults to `false`
    */
-  open var shouldScrollWhenTableViewIsEditing = false
+  open var shouldScrollWhenTableviewIsEditing = false
+
```
[the remark this answers is not in the document]
Comment (StefaniOSApps) [reply Source/ScrollingNavigationController.swift]: whitespace line is ok at this line

File: Source/ScrollingNavigationController.swift [hunk 2]
```diff
@@ -132,6 +132,17 @@ open class ScrollingNavigationController: UINavigationController, UIGestureRecog
...
    */
   open var shouldScrollWhenTableviewIsEditing = false
+ 
+ /**
+   You can hide UIStatusbar at this UIViewController
+   
+   Defaults to `false`
+   */
+  open var shouldPrefersStatusBarHidden = false
+
```
[the remark this answers is not in the document]
Comment (andreamazz) [reply Source/ScrollingNavigationController.swift]: I see no point in this, if the user wants to set the status bar, he should subclass as he would with a common UINavigationController. Also the spelling is wrong

**Comment by andreamazz** [comment]
Hi @StefaniOSApps 
please keep the PRs to a single feature, I can't merge the first part, since it's beyond the scope of this library.
Also, consider squishing the intermediate commits, I'm not too kin in having a bunch of minor commits that fixes trivial spacing and styling problems.

**Comment by StefaniOSApps** [comment]
ok

**Pull request closed without merging**